targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/TARGET_FRDM/PeripheralPinMaps.h

```diff
@@ -0,0 +1,374 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2013 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_PERIPHERALPINMAPS_H
+#define MBED_PERIPHERALPINMAPS_H
+
+#include <mstd_cstddef>
+
+/************GPIO***************/
+MSTD_CONSTEXPR_OBJ_11 PinMap PinMap_GPIO[] = {
```

Comment:
I think we're potentially hitting a problem with multiple tables ending up in ROM in pathological cases, but I'm not sure we can do anything about that in C++14.

If you had two source files each doing

    get_spi_pinmap(pin_a, pin_b);

Where `pin_a` and `pin_b` are variables, then each of those files would have an inline `get_spi_pinmap` which accessed a local `PinMap_SPI`. (`const` objects have internal linkage, so each person including this header gets their own local copy).

Adding `extern` here wouldn't actually work, as it would then end up with multiple definitions.

What we actually need is `inline`, but that's C++17 for objects. Maybe time for a `MSTD_INLINE_OBJ_17`...

Reply:
This may happen only for ARMC5?

Reply:
I think this would be the issue for all compilers. We're safe as long as get_spi_pinmap is always called with constant parameters. 

Fixing it properly requires compilers in C++17 mode so we can add `inline` to the objects.